x/blob/types/payforblob.go

```diff
@@ -331,6 +332,51 @@ func ValidateBlobs(blobs ...*Blob) error {
 	return nil
 }
 
+// validateBlobData returns an error if the size of data is zero or too large.
+func validateBlobData(data []byte) error {
+	if len(data) == 0 {
+		return ErrZeroBlobSize
+	}
+
+	maxBlobSize := blobSizeUpperBound()
+	if len(data) > maxBlobSize {
+		return ErrBlobSizeTooLarge.Wrapf("max blob size is %d bytes", maxBlobSize)
+	}
+	return nil
+}
+
+// blobSizeUpperBound returns an upper bound for the max valid blob size based
+// on the upper bounds for square size and block bytes. Note it is possible that
+// blobs of this size do not fit in a block because the limiting factor may be a
+// parameter in application state (i.e. GovMaxSquareSize). Additionally, even if
+// the application state parameters are at their upper bounds, the number of
+// shares available for blob bytes may be less than estimated in this function
+// (i.e. if the PFB tx shares occupy more than one share). As a result, the
+// upper bound returned by this function may over-estimate the max valid blob
+// size but it should not under-estimate. Consequently, it may be used to
+// immediately reject blobs that are too large but blobs smaller than this upper
+// bound may still fail to be included in a block.
+func blobSizeUpperBound() int {
+	maxSquareSize := appconsts.SquareSizeUpperBound(appconsts.LatestVersion)
```

Comment:
We should pass it but at some point in the calling stack we have to use `appconsts.LatestVersion` because these validation functions are stateless and don't have access to a block's app version

Reply:
https://github.com/celestiaorg/celestia-app/pull/2202/commits/e6248f0961721ef8ec51b073865aef561be0ee65